ocsp/ocsp.go

```diff
@@ -0,0 +1,133 @@
+// Package signer implements certificate signature functionality for CF-SSL.
+package ocsp
+
+import (
+	"bytes"
+	"crypto"
+	"crypto/x509"
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"time"
+
+	cferr "github.com/cloudflare/cfssl/errors"
+	"github.com/cloudflare/cfssl/helpers"
+	"github.com/cloudflare/cfssl/log"
+	"golang.org/x/crypto/ocsp"
+)
+
+var statusCode = map[string]int{
+	"good":    ocsp.Good,
+	"revoked": ocsp.Revoked,
+	"unknown": ocsp.Unknown,
+}
+
+// SignRequest represents the desired contents of a
+// specific OCSP response.
+type SignRequest struct {
+	Certificate *x509.Certificate
+	Status      string
+	Reason      int
+	RevokedAt   time.Time
+}
+
+type Signer interface {
+	Sign(req SignRequest) ([]byte, error)
+}
+
+type StandardSigner struct {
+	issuer    *x509.Certificate
+	responder *x509.Certificate
+	key       crypto.Signer
+	interval  time.Duration
+}
+
+// NewSignerFromFile reads the issuer cert, the responder cert and the responder key
+// from PEM files, and takes an interval in seconds
+func NewSignerFromFile(issuerFile, responderFile, keyFile string, interval int) (Signer, error) {
+	log.Debug("Loading CA cert: ", issuerFile)
```

Comment:
This one is loading the *issuer* cert, the one for whom we're signing OCSP responses.  The responder cert is below.  Changed to "issuer cert".